guides/common/modules/proc_creating-a-playbook-with-modules-from-project-ansible-collections.adoc

```diff
@@ -0,0 +1,102 @@
+[id="creating-a-playbook-with-modules-from-{project-context}-ansible-collections"]
+= Creating a playbook with modules from {Project} Ansible Collections
+
+Ansible modules from the {Project} Ansible Collections must be able to communicate with the {Project} API over HTTPS.
+In your playbooks, include parameters specifying how to authenticate to enable the connection to API.
+
+[IMPORTANT]
+====
+Do not store sensitive credentials, such as username and password, directly in your playbooks or environment variables.
+The following examples use Ansible vault to manage sensitive data.
+====
+
+.Prerequisites
+* A user account in {Project} exists with permissions to perform the action defined in your playbook.
+* You have access to one of the following types of authentication credentials for that user:
+** Username and password
+** Kerberos ticket, if your {ProjectServer} is configured to use {FreeIPA} as an external authentication source
+** {Project} Personal Access Token
+
+.Procedure
+. Store your sensitive variables in an encrypted file:
+.. Create an Ansible vault.
+For example, to create a vault named _My_Vault.yml_:
++
+[subs="+quotes,attributes"]
+----
+$ ansible-vault create _My_Vault.yml_
+----
+.. After the `ansible-vault create` command opens an editor, provide the required parameters in the _key_: _value_ format.
++
+If you want to authenticate with {Project} user credentials:
++
+[source,ini,subs="+quotes,attributes"]
+----
+_My_Server_URL_: _https://{foreman-example-com}_
+_My_Username_: _My_Admin_User_Account_
+_My_Password_: _My_Admin_Password_
+----
++
+If you want to authenticate with a Kerberos ticket or a Personal Access Token:
++
+[source,ini,subs="+quotes,attributes"]
+----
+_My_Server_URL_: _https://{foreman-example-com}_
+----
+.. Save the changes, and close the editor.
+Ansible encrypts the data in the vault.
+. Create a playbook file that references the vault file.
++
+[NOTE]
+====
+To simplify vault file management, the following YAML snippets use module default groups by defining the `module_defaults` parameter.
+With module default groups, any {ansible-namespace} module within the playbook automatically references the vault file, which eliminates the need to pass vault parameters to each module separately.
+====
++
+If you are authenticating with {Project} user credentials:
++
+[source,yaml,subs="+quotes,attributes"]
+----
+- name: _My Playbook_
+  hosts: _localhost_
+  vars_files:
+    - _My_Vault.yml_
+  module_defaults:
+    group/redhat.satellite.satellite:
+      username: "{{ username }}"
+      password: "{{ password }}"
+      server_url: "{{ server_url }}"
+  tasks:
```

Comment:
I could have sworn `tasks: (nothing)` is not a valid playbook, but `ansible-playbook` says it is, TIL!

Reply:
Should I add some kind of placeholder here? I'll try adding at least `- name:`, hopefully it will at least look a bit less weird.

Reply:
fun fact: if you just add `- name:` the sytnax check fails:
```
ERROR! no module/action detected in task.

The error appears to be in '/tmp/tmp.9TR5hUkYtx/lol.yml': line 4, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  tasks:
    - name: test
      ^ here
```

so if you want to add something, it has to be a "reasonable" example (which would just copy the domain example later)

Reply:
Ok, I'm dropping it again. I tried the domain example in one of the earlier drafts but didn't like the repetition (as you're pointing out yourself). I'll keep this thread open to see if I can come up with something. If not, this will probably also do as is.

Reply:
I decided to try adding a new step that explicitly says that users should add their modules under `tasks:`. That required making the structure of the procedure a bit more complex by introducing a new set of substeps, but hopefully it's not overly complex now. Time (and peer review) will tell.